src/main/java/bio/terra/workspace/service/workspace/flight/GoogleBucketFlightMapKeys.java

```diff
@@ -0,0 +1,33 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import bio.terra.workspace.generated.model.GoogleBucketCreationParameters;
+import bio.terra.workspace.generated.model.GoogleBucketDefaultStorageClass;
+import bio.terra.workspace.generated.model.GoogleBucketLifecycle;
+
+/**
+ * Keys into the flight parameter map along with their types. Corresponds to
+ * GoogleBucketCreationParameters
+ */
+public enum GoogleBucketFlightMapKeys {
+  NAME("name", String.class),
+  LOCATION("location", String.class),
+  DEFAULT_STORAGE_CLASS("defaultStorageClass", GoogleBucketDefaultStorageClass.class),
+  LIFECYCLE("lifecycle", GoogleBucketLifecycle.class),
+  BUCKET_CREATION_PARAMS("bucket_creation_params", GoogleBucketCreationParameters.class);
+
+  private final String key;
+  private final Class<?> klass;
+
+  GoogleBucketFlightMapKeys(String key, Class<?> klass) {
+    this.key = key;
+    this.klass = klass;
+  }
+
+  public String getKey() {
+    return key;
+  }
+
+  public Class<? extends Object> getKlass() {
+    return klass;
+  }
```

Comment:
Could you avoid exposing this by making a method like:
```
<T> T getFromMap(FlightMap map, GoogleBucketFlightMapKeys key) {
  return map.get(key.getName(), key.getKlass());
}
```
??

Reply:
What I"d like to do is extend an interface like `FlightMapKey<Class<?> T>` and have several enums that all support lookup into the map. The map class would then support `<T> T get(FlightMapKey<T> key)`. Assuming I can get the enum inheritance to work right. I think I've done it before.

Your flight map  has code I've not seen before but that I think I've been missing around generic casting.

Reply:
I don't think it would work. The type inference gets lost if you're not pointing to a specific member of the enum, and even then, it seems weak.